src/test/java/org/sciborgs1155/robot/HopperTest.java

```diff
@@ -0,0 +1,45 @@
+package org.sciborgs1155.robot;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.sciborgs1155.lib.TestingUtil.*;
+import static org.sciborgs1155.robot.Hopper.HopperConstants.*;
+
+import edu.wpi.first.hal.HAL;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.sciborgs1155.robot.Hopper.Hopper;
+
+public class HopperTest {
+  Hopper hopper = new Hopper();
+
+  @BeforeEach
+  public void setup() {
+    assert HAL.initialize(5000, 0);
+  }
+
+  @Test
+  public void testMovementForward() {
+    run(hopper.forward());
+    assertEquals(MOTOR_MAX_SPEED, hopper.targetSpeed);
+  }
+
+  @Test
+  public void testMovementBackwards() {
+    run(hopper.back());
+    assertEquals(MOTOR_MAX_SPEED * -1, hopper.targetSpeed);
+  }
+
+  @Test
+  public void reachesSetpointWhenMovingForward() {
```

Comment:
similarly, you can combine these two into one movesToSetpoint or reachesSetpoint test, where you set, wait, check, set the other direction, wait some more, check again